packages/strapi-admin/admin/src/containers/LeftMenu/index.js

```diff
@@ -1,202 +1,66 @@
-/*
- *
- * LeftMenu
- *
- */
-
-import React, {
-  forwardRef,
-  memo,
-  useContext,
-  useEffect,
-  useImperativeHandle,
-  useMemo,
-  useReducer,
-} from 'react';
+import React, { memo } from 'react';
 import PropTypes from 'prop-types';
 import { useLocation } from 'react-router-dom';
-
-import { UserContext, hasPermissions, request } from 'strapi-helper-plugin';
 import {
   LeftMenuLinksSection,
   LeftMenuFooter,
   LeftMenuHeader,
   LinksContainer,
 } from '../../components/LeftMenu';
-import { useSettingsMenu } from '../../hooks';
-import { generateModelsLinks, filterLinks } from './utils';
-import init from './init';
-import reducer, { initialState } from './reducer';
 import Loader from './Loader';
 import Wrapper from './Wrapper';
+import useMenuSections from './useMenuSections';
 
-const LeftMenu = forwardRef(({ shouldUpdateStrapi, version, plugins }, ref) => {
+const LeftMenu = ({ shouldUpdateStrapi, version, plugins }) => {
```

Comment:
If the component does not forward the `getModels` ref anymore since it's not needed can you remove the L.257 in the Admin container. The GlobalContextProvider used to expose the getModels from the menu component in order to regenerate the menu. Since now the menu is updated when the permissions changes we don't need it anymore.